Fix event handler and checkoutOneGuestInfo duplication

hucario opened the pull request:
Hitting "edit billing" or "edit shipping" and then cancelling would add another event listener without canceling prior ones. Clicking would trigger all of them at the same time, adding more requests to the server each time until the page refreshed.
Here's a video showing my network requests (cleared after page load) and console to illustrate:
https://files.hucar.io/issues/opc1.mp4

In a similar vein, performing the same cancellation would also duplicate the guest information block (the first time). Here's a video where I've applied a 3px border to `#checkoutOneGuestInfo` to make this clearer:
https://files.hucar.io/issues/opc2.mp4

hucario committed:
Avoid duplication of checkoutOneGuestInfo container

The restoreCustomerInfo endpoint returns a rendered opc_customer_info.php, which does not only include the inner contents of `div#checkoutOneGuestInfo`.
This is then inserted into the existing `div#checkoutOneGuestInfo`, causing an html structure similar to this:
```html
<div id="checkoutOneGuestInfo">
    <div id="checkoutOneGuestInfo">
```
This is obviously invalid, so this commit causes the existing one to be replaced instead.

There are still issues here around how the template also includes comments, and they're duplicated every time this (or what's there already) is run. This isn't great, but-
1. It's not a functional issue the way the previous one was for my CSS
2. It doesn't have an easy answer
so I don't care as much.

hucario committed:
Remove duplicate event listeners

Hitting "edit billing" or "edit shipping" and then cancelling would add another event listener without canceling prior ones. This would multiply the amount of requests to the server each time it occurred until the page refreshed.

hucario commented:
Even with these commits, the HTML comments in the various ajax'd templates are still duplicated.
I just don't have time to fix that right now, and it's fairly low priority since
1. They're completely invisible unless you're looking, and
2. The browser can handle thousands of them without too much trouble, so it would take quite a lot of editing and canceling to trigger issues.

lat9 commented on includes/modules/pages/checkout_one/jquery.checkout_one.js:
```diff
@@ -753,12 +753,8 @@ jQuery(document).ready(function() {
             } else {
                 jQuery(document).on('change', '#checkoutOneBillto input, #checkoutOneBillto select:not(#select-address-bill)', changeBillingFields);
-                jQuery(document).on('click', '#opc-bill-cancel', restoreBilling);
-                jQuery(document).on('click', '#opc-bill-save', saveBilling);
```
Without those 'click' events, how does the shipping/billing address ever get restored or saved?

hucario replied:
There are handlers set when the restore/save shipping/billing functions are defined: 

https://github.com/lat9/one_page_checkout/blob/5df7f8abd2abf61f016ff26380b8156eb900c7cd/includes/modules/pages/checkout_one/jquery.checkout_one.js#L672
https://github.com/lat9/one_page_checkout/blob/5df7f8abd2abf61f016ff26380b8156eb900c7cd/includes/modules/pages/checkout_one/jquery.checkout_one.js#L682
https://github.com/lat9/one_page_checkout/blob/5df7f8abd2abf61f016ff26380b8156eb900c7cd/includes/modules/pages/checkout_one/jquery.checkout_one.js#L721
https://github.com/lat9/one_page_checkout/blob/5df7f8abd2abf61f016ff26380b8156eb900c7cd/includes/modules/pages/checkout_one/jquery.checkout_one.js#L727

Note that you're setting an event handler on the document root and filtering for events from specific elements, rather than setting an event handler on the specific elements directly.
This prevents the handlers from being destroyed when the elements are, as it's not bound to them.

hucario replied:
Yeah. I mention this because if the handlers are not being destroyed in dynamic update, you don't need to add more listeners after you update it. The listeners from before are still there.

hucario committed:
Remove more redundant listeners

These ones matter less, because they aren't causing network requests. The principle remains.

hucario commented:
To reclarify, as I'm interacting again with you on dbio and this bug still exists:
The event listeners are created once on page load, and because they are attached to the document root they are not destroyed when the shipping/billing edit form is destroyed. This is not an issue, and working as intended.

The issue is that `restoreAddressValues` adds listeners for `restoreBilling/restoreShipping` ("restore functions"), which include calls to `restoreAddressValues`. This causes an exponential amount of listeners to be added:
1. Before the first click, there are only the listeners added on page load.
2. The first click triggers a restore function, which calls `restoreAddressValues`, which after an AJAX call adds an additional listener for restore functions. At this point, there are two listeners per button.
3. The second click triggers the restore functions twice, calling `restoreAddressValues` twice, eventually adding two more listeners on top of the previous two. (4 total per button)
4. The third click triggers the restore functions four times, calling `restoreAddressValues` four times, adding four more listeners for 8 total per button.
5. The fourth click leads to 16 total per button, etc

This is what I was attempting to show with [this video](https://files.hucar.io/issues/opc1.mp4), in which you can see first two requests on the first "cancel" click (as expected and fine), then four (!), then eight(!!), and so on.

This also occurs for the `changeShippingFields/changeBillingFields` listeners, but because they don't cause network requests the effect is less noticeable.

My proposed solution (i.e., this pull request) is to avoid creating additional listeners during `restoreAddressValues`.

(Really, this pull should have been two, as the other bug fixed is completely unrelated - but I can't change that now.)
